### PR TITLE
[hail] Thread ExecuteContext through Compile

### PR DESCRIPTION
**Pull request by tpoterba**
stacked on #7535

### CODE REVIEW

**Comment by tpoterba** [comment]
reassigned since Arcturus has a lot of PRs right now and I had non-randomly assigned this based on the dependent PR's assignment anyway

**Comment by tpoterba** [comment]
ready for review

File: hail/src/main/scala/is/hail/HailContext.scala [hunk 1]
```diff
@@ -811,7 +811,7 @@ class HailContext private(
       })
   }
 
-  def readRowsSplit(
+  def readRowsSplit(ctx: ExecuteContext,
```
Comment (chrisvittal) [inline hail/src/main/scala/is/hail/HailContext.scala]: keep consistent style, `ctx` should be on the next line.

File: hail/src/main/scala/is/hail/expr/ir/TableIR.scala [hunk 2]
```diff
@@ -814,7 +814,7 @@ case class TableZipUnchecked(left: TableIR, right: TableIR) extends TableIR {
 
     val rvdType: RVDType = RVDType(inserter.pType, tv1.rvd.typ.key)
 
-    val (t2, makeF) = ir.Compile[Long, Long, Long](
+    val (t2, makeF) = ir.Compile[Long, Long, Long](ctx,
```
Comment (chrisvittal) [inline hail/src/main/scala/is/hail/expr/ir/TableIR.scala]: same style nit

File: hail/src/main/scala/is/hail/expr/ir/TableIR.scala [hunk 3]
```diff
@@ -444,7 +444,7 @@ case class TableFilter(child: TableIR, pred: IR) extends TableIR {
     else if (pred == False())
       return tv.copy(rvd = RVD.empty(HailContext.get.sc, typ.canonicalRVDType))
 
-    val (rTyp, f) = ir.Compile[Long, Long, Boolean](
+    val (rTyp, f) = ir.Compile[Long, Long, Boolean](ctx,
```
Comment (chrisvittal) [inline hail/src/main/scala/is/hail/expr/ir/TableIR.scala]: same style nit

File: hail/src/main/scala/is/hail/expr/ir/TableIR.scala [hunk 4]
```diff
@@ -1008,7 +1008,7 @@ case class TableMapRows(child: TableIR, newRow: IR) extends TableIR {
     var scanSeqNeedsGlobals = false
     var rowIterationNeedsGlobals = false
 
-    val (scanAggs, scanInitOps, scanSeqOps, scanResultType, postScanIR) = ir.CompileWithAggregators[Long, Long, Long](
+    val (scanAggs, scanInitOps, scanSeqOps, scanResultType, postScanIR) = ir.CompileWithAggregators[Long, Long, Long](ctx,
```
Comment (chrisvittal) [inline hail/src/main/scala/is/hail/expr/ir/TableIR.scala]: same style nit

File: hail/src/main/scala/is/hail/expr/ir/TableIR.scala [hunk 5]
```diff
@@ -1020,7 +1020,7 @@ case class TableMapRows(child: TableIR, newRow: IR) extends TableIR {
         seqOp
       })
 
-    val (rTyp, f) = ir.Compile[Long, Long, Long, Long](
+    val (rTyp, f) = ir.Compile[Long, Long, Long, Long](ctx,
```
Comment (chrisvittal) [inline hail/src/main/scala/is/hail/expr/ir/TableIR.scala]: same style nit

File: hail/src/main/scala/is/hail/expr/ir/TableIR.scala [hunk 6]
```diff
@@ -1204,8 +1204,8 @@ case class TableExplode(child: TableIR, path: IndexedSeq[String]) extends TableI
   protected[ir] override def execute(ctx: ExecuteContext): TableValue = {
     val prev = child.execute(ctx)
 
-    val (_, l) = Compile[Long, Int]("row", prev.rvd.rowPType, length)
-    val (t, f) = Compile[Long, Int, Long](
+    val (_, l) = Compile[Long, Int](ctx, "row", prev.rvd.rowPType, length)
+    val (t, f) = Compile[Long, Int, Long](ctx,
```
Comment (chrisvittal) [inline hail/src/main/scala/is/hail/expr/ir/TableIR.scala]: same style nit

**Comment by tpoterba** [comment]
fixed whitespace

**Pull request merged**